pkg/secrethub/audit.go

```diff
@@ -32,3 +33,40 @@ func (c *Client) decryptAuditEvents(events ...*api.Audit) error {
 
 	return nil
 }
+
+func newAuditEventIterator(paginator *http.AuditPaginator, client *Client) AuditEventIterator {
+	return AuditEventIterator{
+		iterator: iterator{
+			paginator: paginator,
+		},
+		paginator:          paginator,
+		decryptAuditEvents: client.decryptAuditEvents,
+	}
+}
+
+type AuditEventIterator struct {
+	iterator
+	decryptAuditEvents func(...*api.Audit) error
+	paginator          *http.AuditPaginator
+}
+
+func (it *AuditEventIterator) Next() (api.Audit, error) {
+	item, err := it.next()
+	if err != nil {
+		return api.Audit{}, err
+	}
+	audit := item.(api.Audit)
```

Comment:
Do we need to check these type casts?

Reply:
`newAuditEventIterator` (line 37) only accepts `*http.AuditPaginator` which always return audit events. As long as the `AuditEventIterator` is created from that constructor, the type check is not necessary.

`AuditEventIterator`s created outside the `secrethub` package, cannot set any of its fields. Does it make sense to build in error checks for such usage?
I believe that would panic on a nil-pointer dereference on a call to next of the `iterator`. The zero-value for the iterator has zero-values for all of its fields. This would mean a value of `nil` for the paginator. Calling `iterator.next` calls `iterator.paginator.Next` which is a nil-pointer reference.